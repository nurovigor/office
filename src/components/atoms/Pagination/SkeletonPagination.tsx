import React from 'react';
import { SkeletonDiv } from 'src/components/atoms/SkeletonDiv';

export const SkeletonPagination = () => {
	return <SkeletonDiv width={200} height={40} />;
};
