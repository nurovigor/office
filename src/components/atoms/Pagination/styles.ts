import { css } from 'linaria';

export const Container = css`
	 {
		display: flex;
		list-style-type: none;
	}
`;

export const Item = css`
	padding: 0 12px;
	height: 32px;
	text-align: center;
	margin: auto 4px;
	color: rgba(0, 0, 0, 0.87);
	display: flex;
	box-sizing: border-box;
	align-items: center;
	letter-spacing: 0.01071em;
	border-radius: 16px;
	line-height: 1.43;
	font-size: 13px;
	min-width: 32px;
	justify-content: center;

	&:hover {
		background-color: rgba(0, 0, 0, 0.04);
		cursor: pointer;
	}
	&.disabled {
		pointer-events: none;
		opacity: 0.5;
	}
`;

export const Dots = css`
	&:hover {
		background-color: transparent;
		cursor: default;
	}
`;

export const Selected = css`
	background-color: rgba(0, 0, 0, 0.08);
`;

export const Arrow = css`
	&::before {
		position: relative;
		content: '';
		display: inline-block;
		width: 3px;
		height: 3px;
		border-right: 0.12em solid rgba(0, 0, 0, 0.87);
		border-top: 0.12em solid rgba(0, 0, 0, 0.87);
	}
`;

export const Left = css`
	transform: rotate(-135deg) translate(-50%);
`;

export const Right = css`
	transform: rotate(45deg);
`;
