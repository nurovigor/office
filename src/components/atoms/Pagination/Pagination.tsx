import React from 'react';
import cn from 'classnames';
import { DOTS, usePagination } from 'src/hooks/usePagination';
import { Container, Item, Dots, Selected, Arrow, Right, Left } from './styles';
import { cx } from '@linaria/core';

type PaginationPropsType = {
	totalCount: number;
	pageSize: number;
	siblingCount?: number;
	currentPage: number;
	// eslint-disable-next-line no-unused-vars
	onPageChange(value: number): void;
};
export const Pagination: React.FC<PaginationPropsType> = ({
	onPageChange,
	totalCount,
	siblingCount = 1,
	currentPage,
	pageSize
}) => {
	const paginationRange = usePagination({
		currentPage,
		totalCount,
		siblingCount,
		pageSize
	});

	if (currentPage === 0 || (paginationRange && paginationRange.length < 2)) {
		return null;
	}

	const onNext = () => {
		onPageChange(currentPage + 1);
	};

	const onPrevious = () => {
		onPageChange(currentPage - 1);
	};

	let lastPage = paginationRange && paginationRange[paginationRange.length - 1];

	return (
		<ul className={Container}>
			<li
				onClick={onPrevious}
				className={cn(Item, {
					disabled: currentPage === 1
				})}
			>
				<div className={cn(Arrow, Left)} />
			</li>
			{paginationRange &&
				paginationRange.map((pageNumber, index) => {
					if (pageNumber === DOTS) {
						return (
							<li key={index} className={cn(Item, Dots)}>
								&#8230;
							</li>
						);
					}

					return (
						<li
							key={index}
							className={cx(Item, currentPage === pageNumber && Selected)}
							onClick={() => onPageChange(Number(pageNumber))}
						>
							{pageNumber}
						</li>
					);
				})}
			<li
				onClick={onNext}
				className={cn(Item, {
					disabled: currentPage === lastPage
				})}
			>
				<div className={cn(Arrow, Right)} />
			</li>
			<div></div>
		</ul>
	);
};
