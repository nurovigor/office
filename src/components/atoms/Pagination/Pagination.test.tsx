import React from 'react';
import { render } from '@testing-library/react';
import { Pagination } from 'src/components/atoms/Pagination';

describe('Tooltip component', () => {
	let callBack = jest.fn();

	it('renders Tooltip', () => {
		const { asFragment } = render(
			<Pagination totalCount={100} pageSize={10} currentPage={1} onPageChange={callBack} />
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
