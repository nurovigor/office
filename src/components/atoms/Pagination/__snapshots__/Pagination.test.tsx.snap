// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip component renders Tooltip 1`] = `
<DocumentFragment>
  <ul>
    <li
      class="disabled"
    >
      <div
        class=""
      />
    </li>
    <li>
      1
    </li>
    <li>
      2
    </li>
    <li>
      3
    </li>
    <li>
      4
    </li>
    <li>
      5
    </li>
    <li
      class=""
    >
      â€¦
    </li>
    <li>
      10
    </li>
    <li
      class=""
    >
      <div
        class=""
      />
    </li>
    <div />
  </ul>
</DocumentFragment>
`;
